src/plugins/discover/public/application/main/state_management/utils/get_state_defaults.test.ts

```diff
@@ -108,7 +115,9 @@ describe('getStateDefaults', () => {
       },
     });
     expect(actualForTextBasedWithValidViewMode.viewMode).toBe(VIEW_MODE.DOCUMENT_LEVEL);
-    expect(actualForTextBasedWithValidViewMode.index).toBe(undefined);
+    expect((actualForTextBasedWithValidViewMode.dataSource as DataViewDataSource).dataViewId).toBe(
+      undefined
+    );
```

Comment:
Can we rather check that `actualForTextBasedWithValidViewMode` is ES|QL data source?